### PR TITLE
feat: configurable timeout and execution interval

### PR DESCRIPTION
**Pull request by someone-stole-my-name**

### CODE REVIEW

**Comment by TwiN** [comment]
I'm fine with this change, but there's a reason why I've been avoiding making it configurable.

By default, cluster-autoscaler will scale down nodes that are older than 10m (or is it 15m?). This means that if the execution interval is too high, it can _significantly_ impact the rolling upgrade time (as this application waits for there to be enough new nodes with sufficient resources before draining, which means that if the execution interval is so long that the new nodes get removed by cluster-autoscaler because it deemed them "unnecessary", it may take significantly longer).

Conversely, if the execution interval is too short, it's possible that the pods drained from the previous execution are not yet ready. 

Long story short, I picked 20s because it was the best middle ground I found. 

---

I'm mostly just leaving it here in case somebody ever starts digging for an explanation.

**Review by TwiN — APPROVED** [review]

**Comment by TwiN** [comment]
Thank you for contributing! If you could confirm on your end that it works as intended, I'd appreciate it :)

**Comment by someone-stole-my-name** [comment]
> Thank you for contributing! If you could confirm on your end that it works as intended, I'd appreciate it :)

Yup, I'm already using this feature 🙇‍♂️

**Comment by sidewinder12s** [comment]
> I'm fine with this change, but there's a reason why I've been avoiding making it configurable.
> 
> By default, cluster-autoscaler will scale down nodes that are older than 10m (or is it 15m?). This means that if the execution interval is too high, it can _significantly_ impact the rolling upgrade time (as this application waits for there to be enough new nodes with sufficient resources before draining, which means that if the execution interval is so long that the new nodes get removed by cluster-autoscaler because it deemed them "unnecessary", it may take significantly longer).
> 
> Conversely, if the execution interval is too short, it's possible that the pods drained from the previous execution are not yet ready.
> 
> Long story short, I picked 20s because it was the best middle ground I found.
> 
> I'm mostly just leaving it here in case somebody ever starts digging for an explanation.

I'll just add I was somewhat interested in this kind of functionality as we started running into AWS API throttling around `Describe*` calls and figured since it takes 2-3 minutes for new nodes to even come up I'd probably be able to naively bump the interval at least up to that.

For older than 10m, its 10m unneeded and 10m after boot by default: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-scale-down-work

**Comment by TwiN** [comment]
Released in [v1.3.0](https://github.com/TwiN/aws-eks-asg-rolling-update-handler/releases/tag/v1.3.0)
